### PR TITLE
[SPARK-22948][k8s] Move SparkPodInitContainer to correct package.

### PR DESCRIPTION
**Pull request by vanzin**

### CODE REVIEW

**Comment by vanzin** [comment]
@foxish @liyinan926

File: dev/sparktestsupport/modules.py [hunk 1]
```diff
@@ -539,7 +539,7 @@ def __hash__(self):
 kubernetes = Module(
     name="kubernetes",
     dependencies=[],
-    source_file_regexes=["resource-managers/kubernetes/core"],
+    source_file_regexes=["resource-managers/kubernetes"],
```
Comment (liyinan926) [inline dev/sparktestsupport/modules.py]: Why this change?

Comment (vanzin) [reply dev/sparktestsupport/modules.py]: Otherwise when you change something outside of kubernetes/core (like the docker files), all unit tests will be run, not just the stuff related to k8s.

I requested this change in the original PR but you guys never addressed it.

Comment (liyinan926) [reply dev/sparktestsupport/modules.py]: Ah, sorry the comment was not caught by us.

**Comment by liyinan926** [comment]
LGTM.

**Comment by vanzin** [comment]
Merging to master / 2.3.
